sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppImpl.java

```diff
@@ -122,8 +122,8 @@ public SpringAppDeployment getActiveDeployment() {
 
     @Override
     @SuppressWarnings("unchecked")
-    public SpringAppDeployments<SpringAppDeploymentImpl> deployments() {
-        return deployments;
+    public <T extends SpringAppDeployment.DefinitionStages.WithCreate<T>> SpringAppDeployments<T> deployments() {
+        return (SpringAppDeployments<T>) deployments;
```

Comment:
If you have better solution for this unchecked warning, please comment. According to @weidongxu-microsoft, it seems the core library also use such way to suppress https://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/Collections.java#l4479